### PR TITLE
Redesign cygrpc.Operation

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**
Further progress toward [issue 12531](https://github.com/grpc/grpc/issues/12531).

### CODE REVIEW

**Comment by mehrdada** [comment]
This PR is against v1.8.x, not master. Is that intentional?

**Comment by nathanielmanistaatgoogle** [comment]
Yes, it's intentional; I'd much rather have it upmerged than have to backport it. :-)

It's not as large as the numbers suggest; a lot of it is reconfiguration rather than new behavior. (Of course, ideally there's zero new non-test behavior in this commit...)

**Comment by mehrdada** [comment]
I agree that upmerge is preferable to backport, but my suggestion was to leave it out of 1.8.x entirely, especially since 1.8 to 1.9 timespan will likely be shorter than usual—if there's no clear bug-fix benefit to 1.8.x users, there's always a risk of some destabilization in a point release, but it's ultimately your call.

**Comment by nathanielmanistaatgoogle** [comment]
I'd like it on `v1.8.x`... I suppose because I'm still thinking about doing a point release once [issue 12531](https://github.com/grpc/grpc/issues/12531) is entirely resolved (or more probably resolved-all-but-for-the-API-change).

File: src/python/grpcio/grpc/_cython/_cygrpc/grpc.pxi [hunk 1]
```diff
@@ -17,6 +17,7 @@ cimport libc.time
 
 # Typedef types with approximately the same semantics to provide their names to
 # Cython
+ctypedef char uint8_t
```
[the remark this answers is not in the document]
Comment (nathanielmanistaatgoogle) [reply src/python/grpcio/grpc/_cython/_cygrpc/grpc.pxi]: Done; thanks for the catch.

**Comment by mehrdada** [comment]
(Reminder for adding the appropriate entry to pony's saddlebag that might be necessary for the new cython file)

**Pull request merged**